dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java

```diff
@@ -1115,18 +1106,24 @@ protected Identifier createIdentifierForFolder(Folder folder, String parentPath)
             sql = sql + " order by " + orderBy;
         }
 
-        dh.setSQLQuery( sql );
-        dh.setFirstResult( offset );
-        dh.setMaxResults( limit );
+        dc.setSQL( sql );
+        dc.setStartRow( offset );
+        dc.setMaxRows( limit );
         if ( identifier.getHostId().equals( Host.SYSTEM_HOST ) ) {
-            dh.setParam( "/" );
-            dh.setParam( identifier.getId() );
+            dc.addParam( "/" );
+            dc.addParam( identifier.getId() );
         } else {
-            dh.setParam( identifier.getURI() + "/" );
-            dh.setParam( identifier.getHostId() );
+            dc.addParam( identifier.getURI() + "/" );
+            dc.addParam( identifier.getHostId() );
         }
 

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)